### PR TITLE
[`tests`] add `accelerate` marker

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR introduces a nice utils for users and contributors (such as myself) that want to run just `accelerate` specific tests. Thanks to @fxmarty that has introduced to me this util

By using `pytest.mark` you can run `accelerate`-specific tests easily. With this PR, if let's say I want to run `accelerate` tests on OPT I can just do:
```bash
RUN_SLOW=1 pytest -m accelerate_tests tests/models/opt/test_modeling_opt.py 
```

cc @ydshieh @sgugger

### CODE REVIEW

**Comment by younesbelkada** [comment]
Thanks a lot for sharing that! 
I agree for consistency it would make sense to do this, however this solution seems IMO much simpler, given the number of affected tests (usually for each model there are only 3 tests), let's see what @sgugger will say!

**Comment by ydshieh** [comment]
> Thanks a lot for sharing that! I agree for consistency it would make sense to do this, however this solution seems IMO much simpler, given the number of affected tests (usually for each model there are only 3 `accelerate` tests)

My response here is not to convince @younesbelkada the suggested approach, but just want to point out the above argument doesn't seem very valid: if you look  `tests/test_modeling_common.py`, there is only one test decorated with `is_pt_tf_cross_test`, and yet we still use that approach.

**Comment by sgugger** [comment]
The difference with the PT/TF cross tests is that we want to skip them unless a special env var is set (to not run them in the torch job and tf job for instance). This is not necessary here since as we don't want to prevent those tests from running in the regular jobs, this is just for convenience.

**Review by sgugger — APPROVED** [review]
Why not, please just make sure to document this marker and the command somewhere (in the test page for instance).

**Review by ydshieh — APPROVED** [review]
OK, thank you for the work and also for Sylvain's explanation.

**Comment by younesbelkada** [comment]
The failing test seems to be unrelated to this PR, merging!

File: docs/source/en/testing.mdx [hunk 1]
```diff
@@ -176,6 +176,15 @@ If you want to include only tests that include both patterns, `and` is to be use
 ```bash
 pytest -k "test and ada" tests/test_optimization.py
 ```
+
+### Run `accelerate` tests
+
+Sometimes you need yo run `accelerate` on your models you can just add `-m accelerate_tests` to your command, if let's say you want to run these tests on `OPT` run:
```
Comment (younesbelkada) [inline docs/source/en/testing.mdx:182]:
```suggestion
Sometimes you need to run `accelerate` tests on your models. For that you can just add `-m accelerate_tests` to your command, if let's say you want to run these tests on `OPT` run:
```

**Pull request merged**